content/docs/buildpack-author-guide/create-buildpack/adding-bill-of-materials.md

```diff
@@ -0,0 +1,171 @@
+
+To add the `ruby` version to the output of `pack inspect-image`, we will have to provide a `Bill-of-Materials` (`BOM`) containing this information. You'll need to update the `launch.toml` created at the end of your `build` script:
+
+```bash
+# ...
+
+# Append a Bill-of-Materials containing metadata about the provided ruby version
+cat >> "$layersdir/launch.toml" <<EOL
+[[bom]]
+name = "ruby"
+[bom.metadata]
+version = "$ruby_version"
+EOL
+```
+
+Your `ruby-buildpack/bin/build` script should look like the following:
+
+<!-- test:file=ruby-buildpack/bin/build -->
+```bash
+#!/usr/bin/env bash
+set -eo pipefail
+
+echo "---> Ruby Buildpack"
+
+# 1. GET ARGS
+layersdir=$1
+plan=$3
+
+# 2. DOWNLOAD RUBY
+rubylayer="$layersdir"/ruby
+mkdir -p "$rubylayer"
+ruby_version=$(cat "$plan" | yj -t | jq -r '.entries[] | select(.name == "ruby") | .metadata.version')
+echo "---> Downloading and extracting Ruby $ruby_version"
+ruby_url=https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/heroku-18/ruby-$ruby_version.tgz
+wget -q -O - "$ruby_url" | tar -xzf - -C "$rubylayer"
+
+# 3. MAKE RUBY AVAILABLE DURING LAUNCH
+echo -e 'launch = true' > "$rubylayer.toml"
+
+# 4. MAKE RUBY AVAILABLE TO THIS SCRIPT
+export PATH="$rubylayer"/bin:$PATH
+export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+${LD_LIBRARY_PATH}:}"$rubylayer/lib"
+
+# 5. INSTALL BUNDLER
+echo "---> Installing bundler"
+gem install bundler --no-ri --no-rdoc
+
+# 6. INSTALL GEMS
+# Compares previous Gemfile.lock checksum to the current Gemfile.lock
+bundlerlayer="$layersdir/bundler"
+local_bundler_checksum=$((sha256sum Gemfile.lock >/dev/null 2>&1 || echo 'DOES_NOT_EXIST') | cut -d ' ' -f 1)
+remote_bundler_checksum=$(cat "$bundlerlayer.toml" | yj -t | jq -r .metadata.checksum 2>/dev/null || echo 'DOES_NOT_EXIST')
+
+if [[ -f Gemfile.lock && $local_bundler_checksum == $remote_bundler_checksum ]] ; then
+    # Determine if no gem dependencies have changed, so it can reuse existing gems without running bundle install
+    echo "---> Reusing gems"
+    bundle config --local path "$bundlerlayer" >/dev/null
+    bundle config --local bin "$bundlerlayer/bin" >/dev/null
+else
+    # Determine if there has been a gem dependency change and install new gems to the bundler layer; re-using existing and un-changed gems
+    echo "---> Installing gems"
+    mkdir -p "$bundlerlayer"
+    cat > "$bundlerlayer.toml" <<EOL
+cache = true
+launch = true
+
+[metadata]
+checksum = "$local_bundler_checksum"
+EOL
+    bundle config set --local path "$bundlerlayer" && bundle install && bundle binstubs --all --path "$bundlerlayer/bin"
+
+fi
+
+# 7. SET DEFAULT START COMMAND
+cat > "$layersdir/launch.toml" <<EOL
+# our web process
+[[processes]]
+type = "web"
+command = "bundle exec ruby app.rb"
+
+# our worker process
+[[processes]]
+type = "worker"
+command = "bundle exec ruby worker.rb"
+EOL
+
+# 8. ADD A BOM
+cat >> "$layersdir/launch.toml" <<EOL
+[[bom]]
+name = "ruby"
+[bom.metadata]
+version = "$ruby_version"
+EOL
+
```

Comment:
Added this minimal BOM to simply replicate what the old BP api would've done before we separated BOM from the build plan. This should help with the concerns that the RFC had about metadata suddenly vanishing.